src/lib/io/sol-uart-impl-riot.c

```diff
@@ -227,3 +227,22 @@ sol_uart_write(struct sol_uart *uart, struct sol_blob *blob)
     }
     return r;
 }
+
+SOL_API int
+sol_uart_get_pending_write_bytes(struct sol_uart *uart, size_t *pending_bytes)
+{
+    int r;
+    struct sol_blob *blob;
+    uint16_t i;
+
+    SOL_NULL_CHECK(uart, -EINVAL);
+    SOL_NULL_CHECK(pending_bytes, -EINVAL);
+
+    *pending_bytes = 0;
+    SOL_PTR_VECTOR_FOREACH_IDX (&uart->pending_blobs, blob, i) {
+        r = sol_util_size_add(*pending_bytes, blob->size, pending_bytes);
+        SOL_INT_CHECK(r, < 0, r);
+    }
+
+    return 0;
+}
```

Comment:
also, what to do regarding partial? Say you have 10 bytes left, arrives 20bytes blob, what do you do?
 - queue the blob anyway, will overflow the allowed memory;
 - reject the blob anyway. Since the data was already produced, then it can be queued at user side or dropped, then it must be produced again.